upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -450,7 +450,9 @@ func (tf *TemplateFunctions) CloudControllerConfigArgv() ([]string, error) {
 		argv = append(argv, fmt.Sprintf("--use-service-account-credentials=%t", true))
 	}
 
-	argv = append(argv, "--cloud-config=/etc/kubernetes/cloud.config")
+	if cluster.Spec.GetCloudProvider() != kops.CloudProviderHetzner {
+		argv = append(argv, "--cloud-config=/etc/kubernetes/cloud.config")
```

Comment:
In most cases, this file is entirely empty. Hetzner CCM does not support this?

Reply:
It supports it, just not mounted as part of the official manifests and didn't want to diverge too much.

Reply:
Maybe consider a PR to upstream